library/os_manila_share.py

```diff
@@ -1,122 +1,65 @@
-    split_export = share['export'].split(":")
-    share['path']= split_export[-1]
-    share['host'] = ':'.join(split_export[:-1])
-    return share
-
-
-def get_access_key(share_client, share_id, user):
-    headers={"X-Openstack-Manila-Api-Version": "2.40"}
-    payload = {"access_list": None}
-    raw_access_list = share_client.post(
-            "/shares/%s/action" % share_id,
-            json=payload, headers=headers).json()['access_list']
-
-    access_key = None
-    for access in raw_access_list:
-        if access['access_to'] == user:
-            access_key = access['access_key']
-            break
-
-    return access_key
-
-
 def main():
     module = AnsibleModule(
         argument_spec = dict(
             name=dict(required=True, type='str'),
-            user=dict(required=True, type='str'),
-            protocol=dict(required=False, type='str'),
-            size_gb=dict(required=False, type='int'), # only checks the size
-            type=dict(required=False, type='str'),  # unused - for creation
+            user=dict(required=False, type='str'),
         ),
         supports_check_mode=False
     )
 
-    share_client = get_share_client(module)
-
+    result = dict(changed=False)
     share_name = module.params['name']
-    share = get_share_details(share_client, share_name)
-
     user = module.params['user']
-    share_id = share['id']
-    access_key = get_access_key(share_client, share_id, user)
-
-    if not access_key:
-        # TODO we could add access here
-        module.fail_json(msg="User does not have access to share.")
-    share['access_key'] = access_key
-
-    required_protocal = module.params.get('protocol')
-    if required_protocal and share['protocol'] != required_protocal:
-        module.fail_json(
-            msg="Protocol %s does not match requested." % share['protocol'])
-
-    required_size = module.params.get('size')
-    if required_size and share['size'] != required_size:
-        # TODO if size doesn't match, we could expand the share
-        module.fail_json(
-            msg="Size %s does not match requested." % share['size'])
-
-    module.exit_json(changed=False, details=share)
+    conn = openstack.connection.from_config()
+    share = conn.share.find_share(share_name)
+    if share is None:
+        module.fail_json('No share named %r found' % share_name)
+    
+    # the share object doesn't actually expose lots of stuff from the API, including the exports
+    share_details = conn.share.get(f"/shares/{share.id}").json()['share']
+    # module.fail_json(share_details) # DEBUG
+    protocol = share_details['share_proto'].upper()
+    if protocol != 'CEPHFS':
+        module.fail_json("Share named %r is protocol %r, can only handle CEPHFS" % (share_name, protocol))
+    
+    # find host and path from CephFS export of form 'mon1:port,mon2:port,mon3:port:path'
+    host, export_path = share_details['export_location'].rsplit(':', 1)
```

Comment:
... I have a feeling the format of this radically changes at one of the ceph release boundaries. But I think this method is still safe as the path is at the end, preceded by a colon still.